packages/hierarchies/src/hierarchies/NodeSelectQueryFactory.ts

```diff
@@ -475,21 +507,46 @@ function createLabelGroupingBaseParamsSelectors(byLabel: ECSqlSelectClauseLabelG
   return selectors;
 }
 
-function createPropertyGroupSelectors(propertyGroup: ECSqlSelectClausePropertyGroup) {
+async function createPropertyGroupSelectors(
+  propertyGroup: ECSqlSelectClausePropertyGroup,
+  fullClassName: string,
+  classLoader: (fullName: string) => Promise<EC.Class>,
+  instanceLabelSelectClauseFactory: IInstanceLabelSelectClauseFactory,
+) {
   const selectors = new Array<{ key: string; selector: string }>();
-  selectors.push(
-    {
-      key: "propertyName",
-      selector: `${createECSqlValueSelector(propertyGroup.propertyName)}`,
-    },
-    {
+  selectors.push({
+    key: "propertyName",
+    selector: `${createECSqlValueSelector(propertyGroup.propertyName)}`,
+  });
+
+  const propertyClass = await classLoader(fullClassName);
+
+  const property = await propertyClass.getProperty(propertyGroup.propertyName);
+  if (!property) {
+    throw new Error(`Property "${propertyGroup.propertyName}" not found in ECClass "${propertyClass.fullName}".`);
+  }
+
+  if (property.isNavigation()) {
+    const customAlias = "cAlias";
+    selectors.push({
+      key: "propertyValue",
+      selector: `(
+          SELECT ${await instanceLabelSelectClauseFactory.createSelectClause({ className: fullClassName, classAlias: customAlias })}
+          FROM ${fullClassName} AS ${customAlias}
+          WHERE [${customAlias}].[ECInstanceId] = [${propertyGroup.propertyClassAlias}].[${propertyGroup.propertyName}].[Id]
+        )`,
```

Comment:
`fullClassName` refers to the class that contains the navigation property. It's not necessarily the same as the target class, e.g. `BisCore.GeometricElement3d.Category` nav prop points to `BisCore.SpatialCategory`.

Reply:
Did not realise that was the case. Changed the implementation to extract target class using `property.relationshipClass.(target/source).abstractConstraint.fullName`. Added tests for checking forward/backward navigation properties